File: shared/ReplicatedStorage/Core/SilhouetteModel.lua
local function blacken(part)
	part.Color = Color3.new(0, 0, 0)
	part.Material = Enum.Material.SmoothPlastic

	if part:IsA("MeshPart") then
		part.TextureID = ""
	end
end

return function(pvInstance)
	if pvInstance:IsA("Part") then
		blacken(pvInstance)
	else
		for _, part in pairs(pvInstance:GetDescendants()) do
			if part:IsA("BasePart") then
				blacken(part)
			end
		end
	end

	return pvInstance
end

File: src/shared/ReplicatedStorage/Core/SilhouetteModel.ts
import { CollectionService } from "@rbxts/services"

function blacken(thing: Instance) {
	if (thing.IsA("BasePart")) {
		thing.Color = new Color3()
		thing.Material = Enum.Material.SmoothPlastic

		if (thing.IsA("MeshPart")) {
			thing.TextureID = ""
		}
	} else if (thing.IsA("Shirt") || thing.IsA("Pants")) {
		thing.Color3 = new Color3()
	} else if (thing.IsA("SpecialMesh")) {
		thing.TextureId = ""
	}
}

function SilhouetteModel(pvInstance: Instance) {
	CollectionService.RemoveTag(pvInstance, "AuroraGun")

	if (pvInstance.IsA("BasePart")) {
		blacken(pvInstance)
	} else if (pvInstance.FindFirstChildOfClass("Shirt")) {
		blacken(pvInstance.FindFirstChildOfClass("Shirt")!)
		blacken(pvInstance.FindFirstChildOfClass("Pants")!)
	} else {
		for (const child of pvInstance.GetDescendants()) {
			blacken(child)
		}
	}
}

export = SilhouetteModel
